app/pages/preview/contextCreator.js

```diff
@@ -58,7 +58,73 @@ const generateCallOffAndSupplierDetailsTable = ({
 
 const getCurrentDate = () => formatDate(new Date(Date.now()));
 
-export const getContext = ({ orderId, orderData }) => ({
+const generateRecurringCostDetailsTable = ({
+  recurringCostTable, recurringCostItems = [], serviceRecipients = {},
+}) => {
+  const items = recurringCostItems.map((item) => {
+    const classes = 'nhsuk-u-font-size-14';
+    const columns = [];
+
+    if (!Object.prototype.hasOwnProperty.call(serviceRecipients, item.serviceRecipientsOdsCode)) {
```

Comment:
think you can shorthand this too

`if (!serviceRecipients[item.serviceRecipientsOdsCode])`